src/libertem/common/math.py

```diff
@@ -0,0 +1,15 @@
+try:
+    from math import prod
+except ImportError:
+    def prod(iterable):
+        '''
+        Safe product for large size calculations.
+
+        :meth:`numpy.prod` uses 32 bit for default :code:`int` on Windows 64 bit. This
+        function is a replacement for :meth:`math.prod` for versions < Python 3.8
+        that uses infinite width integers.
+        '''
+        result = 1
+        for item in iterable:
+            result *= item
```

Comment:
Should the `item` be cast explicitly to `int` here? Because `iterable` could also contain `numpy` types, which means `np.ndarray.__mul__` will be used, correct? How does `math.prod` work in that case? It would be nice to have a type annotation here, too (right now, that would be `Iterable[int]`, if I'm not mistaken).

Maybe we do need some basic unit tests for this?